x-pack/test/ui_capabilities/common/nav_links_builder.ts

```diff
@@ -26,21 +26,24 @@ export class NavLinksBuilder {
   public all() {
     return this.build(() => true);
   }
-  public except(...feature: string[]) {
-    return this.build((featureId) => !feature.includes(featureId));
+  public except(features: string[], apps: string[] = []) {
```

Comment:
__note:__ Even though providing `apps` this way is simpler, I think it's not quite correct since it makes it seem like features and apps are on the same "level", while in reality, apps always belong to a specific feature. So I'd suggest changing it to something like this instead. While it's more complex, it's more accurate and also won't require any changes to the existing code that doesn't deal with enterprise search features and apps (and the similar code for `only` function):
```ts
  public except(...features: Array<string | { feature: string; apps: [] }>) {
    return this.build((featureId, appId) =>
      features.some((feature) =>
        typeof feature === 'string'
          ? featureId !== feature
          : feature.feature !== featureId || !feature.apps.includes(appId)
      )
    );
  }
.....
expect(uiCapabilities.value!.navLinks).to.eql(
  navLinksBuilder.except('monitoring', { feature: 'enterpriseSearch', apps: ['appSearch', 'workplaceSearch']})
);
```

What do you think?

Reply:
Fixed: https://github.com/elastic/kibana/pull/192130/commits/099825e519c5fe5624dbac00a4fe6782a8c8bd3b